Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp

```diff
@@ -426,14 +466,137 @@ Optional<String> to_smallest_temporal_unit(GlobalObject& global_object, Object&
+        return;
+    }
+    // 7. If smallestUnit is "second" and largestUnit is "millisecond", "microsecond", or "nanosecond", then
+    if (smallest_unit == "second"sv && largest_unit.is_one_of("millisecond"sv, "microsecond"sv, "nanosecond"sv)) {
+        // a. Throw a RangeError exception.
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidUnitRange, smallest_unit, largest_unit);
+        return;
+    }
+    // 8. If smallestUnit is "millisecond" and largestUnit is "microsecond" or "nanosecond", then
+    if (smallest_unit == "millisecond"sv && largest_unit.is_one_of("microsecond"sv, "nanosecond"sv)) {
+        // a. Throw a RangeError exception.
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidUnitRange, smallest_unit, largest_unit);
+        return;
+    }
+    // 9. If smallestUnit is "microsecond" and largestUnit is "nanosecond", then
+    if (smallest_unit == "microsecond"sv && largest_unit == "nanosecond"sv) {
+        // a. Throw a RangeError exception.
+        vm.throw_exception<RangeError>(global_object, ErrorType::TemporalInvalidUnitRange, smallest_unit, largest_unit);
+        return;
+    }
+}
+
+// 13.23 LargerOfTwoTemporalUnits ( u1, u2 ), https://tc39.es/proposal-temporal/#sec-temporal-largeroftwotemporalunits
+String larger_of_two_temporal_units(String const& unit1, String const& unit2)
+{
+    // 1. If either u1 or u2 is "year", return "year".
+    if (unit1 == "year"sv || unit2 == "year"sv)
+        return "year"sv;
+    // 2. If either u1 or u2 is "month", return "month".
+    if (unit1 == "month"sv || unit2 == "month"sv)
+        return "month"sv;
+    // 3. If either u1 or u2 is "week", return "week".
+    if (unit1 == "week"sv || unit2 == "week"sv)
+        return "week"sv;
+    // 4. If either u1 or u2 is "day", return "day".
+    if (unit1 == "day"sv || unit2 == "day"sv)
+        return "day"sv;
+    // 5. If either u1 or u2 is "hour", return "hour".
+    if (unit1 == "hour"sv || unit2 == "hour"sv)
+        return "hour"sv;
+    // 6. If either u1 or u2 is "minute", return "minute".
+    if (unit1 == "minute"sv || unit2 == "minute"sv)
+        return "minute"sv;
+    // 7. If either u1 or u2 is "second", return "second".
+    if (unit1 == "second"sv || unit2 == "second"sv)
+        return "second"sv;
+    // 8. If either u1 or u2 is "millisecond", return "millisecond".
+    if (unit1 == "millisecond"sv || unit2 == "millisecond"sv)
+        return "millisecond"sv;
+    // 9. If either u1 or u2 is "microsecond", return "microsecond".
+    if (unit1 == "microsecond"sv || unit2 == "microsecond"sv)
+        return "microsecond"sv;
+    // 10. Return "nanosecond".
+    return "nanosecond"sv;
+}
+
+// 13.25 MaximumTemporalDurationRoundingIncrement ( unit ), https://tc39.es/proposal-temporal/#sec-temporal-maximumtemporaldurationroundingincrement
+Optional<u16> maximum_temporal_duration_rounding_increment(String const& unit)
```

Comment:
Again, can be a StringView passed by value (we have a couple of those across temporal where it's unnecessary, but I'd like to move away from that)

Reply:
fixed